### ISSUE TITLE
when editing vlans with the api, vlans without a group are considered as one group, resulting in duplicate-errors

### ISSUE THREAD

**Issue by oehTie** [netbox-community/netbox#17979]
### Deployment Type

Self-hosted

### Triage priority

N/A

### NetBox Version

v4.1.6

### Python Version

3.10

### Steps to Reproduce

Using the API, create a vlan without a group, 
using the api, create a second vlan with the same vlan-id, this works.
using the api, create a third vlan with a different vlan-id, also without a group, this works. 
using the api, change the first vlan to have the same vlan id as the second one, still without a group. This results in an error that the vlan is duplicate.

### Expected Behavior

as the vlans have no group, duplicates should be allowed. 

When the same steps to reproduce are executed using the Netbox Gui, it works as expected.

### Observed Behavior

Error: The fields group, vid must make a unique set.

exact api calls as already posted:"

@jeremystretch here is the API call flow that will result in the error:

POST ungrouped VLAN with vid 1000 -> Success with ID 1

curl -X 'POST' \
  'http://localhost:8000/api/ipam/vlans/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"comments":"","description":"","group":null,"name":"VLAN-1","role":1,"status":"active","vid":1000}'

POST ungrouped VLAN with vid 1000 -> Success with ID 2

curl -X 'POST' \
  'http://localhost:8000/api/ipam/vlans/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"comments":"","description":"","group":null,"name":"VLAN-2","role":1,"status":"active","vid":1000}'
POST ungrouped VLAN with vid 1001 -> Success with ID 3

curl -X 'POST' \
  'http://localhost:8000/api/ipam/vlans/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"comments":"","description":"","group":null,"name":"VLAN-3","role":1,"status":"active","vid":1001}'
PUT VLAN with ID 1 to update VID to 1001 -> Error: Bad request: {"non_field_errors": ["The fields group, vid must make a unique set."]}

curl -X 'PUT' \
  'http://localhost:8000/api/ipam/vlans/1/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"comments":"","description":"","group":null,"name":"VLAN-1","role":1,"status":"active","vid":1001}'
PATCH VLAN with ID 1 to update VID to 1001 -> Error: Bad request: {"non_field_errors": ["The fields group, vid must make a unique set."]}

curl -X 'PATCH' \
  'http://localhost:8000/api/ipam/vlans/1/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"vid":1001}'

**Comment by oehTie** [comment]
the exact api call:


curl -X 'PUT' \
  'http://localhost:8000/api/ipam/vlans/4060/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token *****************************' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ********************************************' \
  -d '{"comments":"","description":"","group":null,"interfaces":[],"name":"VLAN-3","role":3,"status":"active","vid":1000}'

response:

Error: Bad request: {
  "non_field_errors": [
    "The fields group, vid must make a unique set."
  ]
}

**Comment by jeremystretch** [comment]
Please edit your post above to include the *specific values* and REST API query used in the creation of each object. This is necessary for someone to attempt to reproduce the behavior.

**Comment by jorritberendsen-forefreedom** [comment]
@jeremystretch here is the API call flow that will result in the error:

# POST ungrouped VLAN with vid 1000 -> Success with ID 1
```
curl -X 'POST' \
  'http://localhost:8000/api/ipam/vlans/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"comments":"","description":"","group":null,"name":"VLAN-1","role":1,"status":"active","vid":1000}'

```
# POST ungrouped VLAN with vid 1000 -> Success with ID 2
```
curl -X 'POST' \
  'http://localhost:8000/api/ipam/vlans/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"comments":"","description":"","group":null,"name":"VLAN-2","role":1,"status":"active","vid":1000}'
```

# POST ungrouped VLAN with vid 1001 -> Success with ID 3
```
curl -X 'POST' \
  'http://localhost:8000/api/ipam/vlans/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"comments":"","description":"","group":null,"name":"VLAN-3","role":1,"status":"active","vid":1001}'
```

# PUT VLAN with ID 1 to update VID to 1001 -> Error: Bad request: {"non_field_errors": ["The fields group, vid must make a unique set."]}
```
curl -X 'PUT' \
  'http://localhost:8000/api/ipam/vlans/1/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"comments":"","description":"","group":null,"name":"VLAN-1","role":1,"status":"active","vid":1001}'
```

# PATCH VLAN with ID 1 to update VID to 1001 -> Error: Bad request: {"non_field_errors": ["The fields group, vid must make a unique set."]}
```
curl -X 'PATCH' \
  'http://localhost:8000/api/ipam/vlans/1/' \
  -H 'accept: application/json' \
  -H 'Authorization: Token ***' \
  -H 'Content-Type: application/json' \
  -H 'X-CSRFTOKEN: ***' \
  -d '{"vid":1001}'
```

**Comment by oehTie** [comment]
info added to starting post

**Comment by jeremystretch** [comment]
This seems like another instance of the root issue recently resolved under bug #17810, which should be fixed in the next release (v4.1.8). I'm going to mark this as resolved, but please feel free to ask that it be re-opened if you're able to reproduce the behavior on v4.1.8 once it has been released.
